scriptworker/github.py

```diff
@@ -114,8 +117,15 @@ def get_tag_hash(self, tag_name):
         repo = self._github_repository.html_url
 
         url = '/'.join([repo.rstrip('/'), 'branch_commits', revision])
-        html_data = await retry_request(context, url)
-        html_text = html_data.strip()
+        from scriptworker.task import get_decision_task_id
+        cache_key = '{}-{}'.format(get_decision_task_id(context.task), url)
```

Comment:
I think we might eventually hit an issue because the cache isn't invalidated over time, or because the dictionary takes too much memory. We may want to eventually use https://github.com/tkem/cachetools.

That said, the patch looks good enough to me for now. I'll deploy it and see if fixes the immediate issues we're seeing.